payjoin-directory/src/lib.rs

```
@@ -341,11 +344,11 @@ async fn post_fallback_v1(
         .await
         .map_err(|e| HandlerError::BadRequest(e.into()))?;
     match pool.peek_v1(id).await {
-        Some(result) => match result {
-            Ok(buffered_req) => Ok(Response::new(full(buffered_req))),
-            Err(e) => Err(HandlerError::BadRequest(e.into())),
+        Ok(buffered_req) => Ok(Response::new(full(buffered_req))),
+        Err(e) => match e {
+            DbPoolError::Redis(_) => Err(HandlerError::BadRequest(e.into())),
```

IF we want to return the Redis error to the client (big IF, @nothingmuch what do you think) I think we can return Redis(re) => Err(HandlerError::BadRequest(re.into()) rather than the top level error, since that's the only returned variant

---

Is the concern here that Redis will be replaced in the near future (#419), or just generally that the client shouldn't have to worry about the underlying implementation?

---

exposing internals of the server

1. isn't generally actionable by most callers
2. should just be error logged instead and
3. expose details about the server state that potentially could be used for an attack. though I'm not sure I can think of one off the top of my head, my idea of best practice is to reveal only that necessary and actionable serverstate for the client to operate and keep everything else private.